web3c/oasis.js

```diff
@@ -17,15 +17,23 @@ const makeContractFactory = require('./contract_factory');
  */
 class Oasis {
 
-  constructor(web3, storage, mraebox) {
-    this.keyManager = new KeyManager(web3, storage, mraebox);
+  constructor(web3, storage, mraebox, options) {
     this.utils = new OasisUtils();
+    this._setupKeyManager(web3, storage, mraebox, options);
     this._setupRpcs(web3);
     this._setupContract(web3, storage, mraebox);
     this._setupWrappedAccounts(web3);
     this._setupSubscribe(web3);
   }
 
+  _setupKeyManager(web3, storage, mraebox, options) {
+    let signer = undefined;
```

Comment:
https://github.com/oasislabs/web3c.js/blob/cb72f4034db7318cc71f3ad1cb58f6d9691537df/web3c/key_manager.js#L28
:ok_hand:

Reply:
Indeed, `signer` is an optional argument in the key manager already. It is currently used to inject signature validation for tests.